docs/USAGE-GUIDE.md

```diff
@@ -536,6 +568,53 @@ callback can get any ClientHello infromation from the connection and use
 The callback can return 0 to continue handshake in s2n or it can return negative
 value to make s2n terminate handshake early with fatal handshake failure alert.
 
+## Client Auth Related calls
+
+
+### s2n\_config\_set\_client\_auth\_type and s2n\_connection\_set\_client\_auth\_type
+```c
+int s2n_config_set_client_auth_type(struct s2n_config *config, s2n_cert_auth_type cert_auth_type);
+int s2n_connection_set_client_auth_type(struct s2n_connection *conn, s2n_cert_auth_type cert_auth_type);
+```
+Sets whether or not a Client Certificate should be required to complete the TLS Connection.
+If this is set to **S2N_CERT_AUTH_REQUIRED** then a **verify_cert_trust_chain_fn** callback should be provided as well since the current
+default is for s2n to accept all RSA Certs on the client side, and deny all certs on the server side.
+
+
+### verify_cert_trust_chain_fn
+
+```c
+int verify_cert_trust_chain(uint8_t *der_cert_chain_in, uint32_t cert_chain_len, struct s2n_cert_public_key *public_key_out, void *context);
+```
+ - **der_cert_chain_in** The DER encoded full chain of certificates recieved
+ - **cert_chain_len** The length in bytes of the DER encoded Cert Chain
+ - **public_key_out** The public key that should be updated with the key extracted from the first certificate in the chain (the leaf Cert)
+ - **context** A pointer to any caller defined context data needed for the callback (Cert Trust Store, etc)
+ - **return Value** The function should return 0 if the Certificate Chain is trusted and public key extraction was successful, and less than 0 if any Certificate in the chain is untrusted, or there was some other error.
+
+**verify_cert_trust_chain_fn** defines a Callback Function Signature intended to be used only in special circumstances, and may be removed in a later release.
+Implementations should Verify the Certificate Chain of trust, and place the leaf Certificate's Public Key in the public_key_out parameter.
+They should not not perform any hostname validation, which is still needed in order to completely validate a Certificate.
+
+### Public Key API's
+```c
+int s2n_rsa_public_key_set_from_openssl(struct s2n_rsa_public_key *s2n_rsa, RSA *openssl_rsa);
```

Comment:
I'm open to ideas, but I don't know of a better way. We do have those functions in `crypto/s2n_rsa.c` but almost all of them operate on `s2n_blob` which is not exported as part of the API and I don't believe we want to export that struct.